### PR TITLE
Remove legacy rendering code

### PR DESCRIPTION
**Pull request by cepetr**
"This PR finalizes #3695 and removes legacy (unused) code from the Rust and C codebase that has not been used since the `NEW_RENDERING` option was enabled everywhere.

Key changes:

1) `NEW_RENDERING` option has been fully removed. All code, including boardloaders and bootloaders, now uses the new drawing library and/or `gfx/bitblt` functions.
2) Legacy display drivers located in the `/displays` folder have been removed.
3) The `fn paint()` method has been removed from the `Component` interface; now we only use `fn render()`.

Note: The new display drivers still reside in the `xdisplay` folder and will be relocated in a subsequent PR.

### CODE REVIEW

File: core/embed/lib/gfx_draw.h [hunk 1]
```diff
@@ -141,22 +144,26 @@ void gfx_draw_bar(gfx_rect_t rect, gfx_color_t color);
 // Not all bitmap formats are supported now. Please see the implementation.
 void gfx_draw_bitmap(gfx_rect_t rect, const gfx_bitmap_t* bitmap);
 
+#define GFX_ALIGN_LEFT 0x00    // Align text to the left (horizontal)
```
Comment (TychoVrahe) [inline core/embed/lib/gfx_draw.h:147]: may be better as enum

File: core/embed/lib/gfx_draw.h [hunk 2]
```diff
@@ -141,22 +144,26 @@ void gfx_draw_bar(gfx_rect_t rect, gfx_color_t color);
...
+#define GFX_ALIGN_CENTER 0x01  // Align text to the center (horizontal)
+
 // Draws a text to the specified position.
 //
 // `offset` - the most left point on the font baseline
 // `text` - utf-8 text
 // `maxlen` - maximum number of characters displayed (use SIZE_MAX when not
 // specified) `attr` - font & text color
+// `optionsz - combination of GFX_ALIGN_xxx flags
```
Comment (TychoVrahe) [inline core/embed/lib/gfx_draw.h:156]: outdated comment

File: core/embed/prodtest/main.c [hunk 3]
```diff
@@ -193,15 +200,22 @@ void extract_params(const char *str, int *numbers, int *count, int max_count) {
...
+  gfx_draw_bar(r_out, COLOR_WHITE);
+  gfx_draw_bar(r_in, COLOR_BLACK);
+
   display_refresh();
 }
 
 static void draw_welcome_screen(void) {
 #if defined TREZOR_MODEL_R || defined TREZOR_MODEL_T3B1
-  display_bar(0, 0, DISPLAY_RESX, DISPLAY_RESY, 0xFFFF);
+  gfx_draw_bar(gfx_rect_wh(0, 0, DISPLAY_RESX, DISPLAY_RESY), 0xFFFF);
```
Comment (TychoVrahe) [inline core/embed/prodtest/main.c:218]: lets use COLOR_WHITE, otherwise it won't work for 32bit colors

File: core/embed/prodtest/main.c [hunk 4]
```diff
@@ -235,7 +249,9 @@ static void test_display(const char *colors) {
         c = 0xFFFF;
```
Comment (TychoVrahe) [inline core/embed/prodtest/main.c:249]: also in this functions, proper color constants are needed to ensure function across different color depths.

File: core/embed/prodtest/main.c [hunk 5]
```diff
@@ -365,19 +381,19 @@ static void test_touch(const char *args) {
   const int width = DISPLAY_RESX / 2;
   const int height = DISPLAY_RESY / 2;
 
-  display_clear();
+  gfx_clear();
   switch (column) {
     case 1:
-      display_bar(0, 0, width, height, 0xFFFF);
+      gfx_draw_bar(gfx_rect_wh(0, 0, width, height), 0xFFFF);
```
Comment (TychoVrahe) [inline core/embed/prodtest/main.c:387]: also here

File: core/embed/prodtest/main.c [hunk 6]
```diff
@@ -421,8 +437,8 @@ static void test_touch_custom(const char *args) {
 
   uint32_t ticks_start = hal_ticks_ms();
 
-  display_clear();
-  display_bar(x, y, width, height, 0xFFFF);
+  gfx_clear();
+  gfx_draw_bar(gfx_rect_wh(x, y, width, height), 0xFFFF);
```
Comment (TychoVrahe) [inline core/embed/prodtest/main.c:441]: and here

File: core/embed/prodtest/main.c [hunk 7]
```diff
@@ -538,19 +554,19 @@ static void test_sensitivity(const char *args) {
...
 
   for (;;) {
     uint32_t evt = touch_get_event();
     if (evt & TOUCH_START || evt & TOUCH_MOVE) {
       int x = touch_unpack_x(evt);
       int y = touch_unpack_y(evt);
-      display_clear();
-      display_bar(x - 48, y - 48, 96, 96, 0xFFFF);
+      gfx_clear();
+      gfx_draw_bar(gfx_rect_wh(x - 48, y - 48, 96, 96), 0xFFFF);
```
Comment (TychoVrahe) [inline core/embed/prodtest/main.c:566]: here too

File: core/site_scons/models/D002/discovery2.py [hunk 8]
```diff
@@ -76,25 +69,20 @@ def configure(
 
     defines += [
         "USE_DMA2D",
-        "FRAMEBUFFER",
         "FRAMEBUFFER32BIT",
```
Comment (TychoVrahe) [inline core/site_scons/models/D002/discovery2.py:72]: not needed anymore

File: core/SConscript.unix [hunk 9]
```diff
@@ -415,16 +401,6 @@ SOURCE_UNIX = [
...
 ]
 
-if NEW_RENDERING:
-    SOURCE_MOD += [
-        'embed/trezorhal/unix/display_driver.c',
-        'embed/trezorhal/xdisplay_legacy.c',
-    ]
-else:
-    SOURCE_MOD += [
-        'embed/trezorhal/unix/display-unix.c',
```
Comment (TychoVrahe) [inline core/SConscript.unix:425]: this file should be deleted

**Review by TychoVrahe — CHANGES_REQUESTED** [review]
feels great to get rid of this finally

File: core/site_scons/models/D002/discovery2.py (hunk 8 above)
Comment (TychoVrahe) [reply core/site_scons/models/D002/discovery2.py:72]: or rather, we need FRAMEBUFFER instead of FRAMEBUFFER32BIT

**Comment by TychoVrahe** [comment]
The xdisplay folder in trezorhal/stm32xx should probably be renamed. But maybe that is better left to be handled in the successive reorganization PR?

File: core/embed/lib/gfx_draw.h (hunk 1 above)
Comment (cepetr) [reply core/embed/lib/gfx_draw.h]: yes, it's better, fixed, see 4c1cd9be909859df36e3e62711dd69ebd25b3c3f

File: core/embed/lib/gfx_draw.h (hunk 2 above)
Comment (cepetr) [reply core/embed/lib/gfx_draw.h]: fixed, see 4c1cd9be909859df36e3e62711dd69ebd25b3c3f

File: core/embed/prodtest/main.c (hunk 3 above)
Comment (cepetr) [reply core/embed/prodtest/main.c]: fixed, see 989fffebf2196106384103bd7800dfdcbb7a54b9

File: core/embed/prodtest/main.c (hunk 4 above)
Comment (cepetr) [reply core/embed/prodtest/main.c]: fixed, see 989fffebf2196106384103bd7800dfdcbb7a54b9

File: core/embed/prodtest/main.c (hunk 5 above)
Comment (cepetr) [reply core/embed/prodtest/main.c]: fixed, see 989fffebf2196106384103bd7800dfdcbb7a54b9

File: core/embed/prodtest/main.c (hunk 6 above)
Comment (cepetr) [reply core/embed/prodtest/main.c]: fixed, see 989fffebf2196106384103bd7800dfdcbb7a54b9

File: core/site_scons/models/D002/discovery2.py (hunk 8 above)
Comment (cepetr) [reply core/site_scons/models/D002/discovery2.py]: I removed FRAMEBUFFER32BIT. FRAMEBUFFER is defined few lines below. See 0b1c85d7a7e21a583c521d8061a32f0b1901a0ca

File: core/SConscript.unix (hunk 9 above)
Comment (cepetr) [reply core/SConscript.unix:425]: deleted, see 0b1c85d7a7e21a583c521d8061a32f0b1901a0ca

File: core/embed/prodtest/main.c (hunk 7 above)
Comment (cepetr) [reply core/embed/prodtest/main.c]: fixed, see 989fffebf2196106384103bd7800dfdcbb7a54b9

**Review by TychoVrahe — APPROVED** [review]

**Review by matejcik — APPROVED** [review]

**Pull request merged**